lib/core/src/Cardano/Pool/DB/Sqlite.hs

```diff
@@ -383,16 +370,27 @@ newDBLayer trace fp timeInterpreter = do
             deleteWhere [ PoolRetirementSlot >. point ]
             -- TODO: remove dangling metadata no longer attached to a pool
 
-        , removePools = mapM_ $ \pool -> do
-            liftIO
-                $ traceWith trace
-                $ MsgRemovingDatabaseEntity
-                $ T.unwords [ "pool", toText pool ]
-            deleteWhere [ PoolProductionPoolId ==. pool ]
-            deleteWhere [ PoolOwnerPoolId ==. pool ]
-            deleteWhere [ PoolRegistrationPoolId ==. pool ]
-            deleteWhere [ PoolRetirementPoolId ==. pool ]
-            deleteWhere [ StakeDistributionPoolId ==. pool ]
+        , removePools = removePools_
+
+        , removeRetiredPools = \epoch -> do
+            let report = liftIO . traceWith trace . MsgGarbageCollection
+            report $ T.concat
+                [ "Looking for pools that retired in or before epoch "
+                , toText epoch
+                , "."
+                ]
+            listRetiredPools_ epoch >>= \case
+                [] -> do
+                    report "Found no retired pools."
+                    pure []
+                retirementCerts -> do
+                    report $ T.unlines
```

Comment:
Part of the benefit of typed tracing is that you don't need to obscure the code with log string formatting.

I would make two messages:
 - `MsgRemoveRetiredPools EpochNumber BracketLog` (covers start and finish - use with `bracketTracer`)
 - `MsgRemoveRetiredPoolsCerts [Cert]`

Then put all the message formatting in the ToText instance.

Reply:
@rvl wrote:
> Part of the benefit of typed tracing is that you don't need to obscure the code with log string formatting.

Thanks for this feedback.

I do agree with you. However, one issue is that `DBLog` type is general to all databases. Ideally, `DBLog` shouldn't use types that are specific to _pools_ or _wallets_. 

Perhaps one solution would be to introduce `WalletDbLog` and `PoolDbLog` types, of which the latter would have the constructors that you suggest: `MsgRemoveRetiredPools` and so on. However, this would require changing references to `DBLog` in quite a few places. If we decide to do this, I suggest we create a separate PR (or make an item of technical debt), in order to avoid complicating this one too much.

> I would make two messages:
>   * `MsgRemoveRetiredPools EpochNumber BracketLog` (covers start and finish - use with `bracketTracer`)
>   * `MsgRemoveRetiredPoolsCerts [Cert]`

Thanks for this suggestion: I didn't know about `bracketTracer`.

I've adjusted `removeRetiredPools` to use bracket-style tracing, with the following general constructors (defined within `DBLog`):

```hs
| MsgGarbageCollection Text BracketLog
| MsgGarbageCollectionStep Text
```

See: c912a269501343563283281b5771c66b619e5d32

Reply:
I've created a technical debt ticket to cover the idea of defining separate `WalletDbLog` and `PoolDbLog` types here: https://jira.iohk.io/browse/ADP-407.

While tackling the above ticket, we could also go through the existing `DbLog` constructors and replace the "stringly-typed" constructors with strongly-typed equivalents in `WalletDbLog` and `PoolDbLog`. Hopefully, this should also clean up parts of the code that are obscured with log formatting logic.

Reply:
@rvl we now have the following types:

https://github.com/input-output-hk/cardano-wallet/blob/ab68664b81304faaa3037ce5c1e7a772f555ba45/lib/core/src/Cardano/Pool/DB/Log.hs#L35-L36